libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -2959,18 +2959,27 @@ void GCS_MAVLINK::handle_common_vision_position_estimate_data(const uint64_t use
                                                               const float roll,
                                                               const float pitch,
                                                               const float yaw,
+                                                              const float covariance[21],
                                                               const uint8_t reset_counter,
                                                               const uint16_t payload_size)
 {
 #if HAL_VISUALODOM_ENABLED
+    float posErr = 0;
+    float angErr = 0;
     // correct offboard timestamp to be in local ms since boot
     uint32_t timestamp_ms = correct_offboard_timestamp_usec_to_ms(usec, payload_size);
 
     AP_VisualOdom *visual_odom = AP::visualodom();
     if (visual_odom == nullptr) {
         return;
     }
-    visual_odom->handle_vision_position_estimate(usec, timestamp_ms, x, y, z, roll, pitch, yaw, reset_counter);
+
+    if (!isnan(covariance[0])) {
+        posErr = cbrtf(sq(covariance[0])+sq(covariance[6])+sq(covariance[11]));
```

Comment:
@chobitsfan @rmackay9 Hi. This approximation of a mean std.dev over all axes seems wrong to me. For a cov.matrix with all diagonal elements equal to 2, it gives 2.3, while correct answer is 1.41.
It'd look more reasonable to remove squares (gives 1.82)
```posErr = cbrtf(covariance[0]+covariance[6]+covariance[11]);```
or to use square root (gives 1.41).
```posErr = sqrtf((covariance[0]+covariance[6]+covariance[11])/3);```